addons/stock/wizard/product_label_layout.py

```diff
@@ -37,13 +37,13 @@ def _prepare_report_data(self):
         elif self.move_quantity == 'move' and self.move_ids.move_line_ids:
             custom_barcodes = defaultdict(list)
             for line in self.move_ids.move_line_ids:
-                if line.product_uom_id.category_id == uom_unit:
+                if line.product_uom_id.labellable:
                     if (line.lot_id or line.lot_name) and int(line.quantity):
                         custom_barcodes[line.product_id.id].append((line.lot_id.name or line.lot_name, int(line.quantity)))
                         continue
                     quantities[line.product_id.id] += line.quantity
                 else:
-                    quantities[line.product_id.id] = 1
+                    quantities[line.product_id.id] += len(line)
```

Comment:
On validating a reception of 2 kilos of prod_A, you then on changing the Quantity to 5 =>  now have 1 stock move lines instead of 2. Demo- https://shorturl.at/YziuM I have tried also for multistep, Am I missing something?

Reply:
Ok, maybe I didn't explain correctly:
- you create a reception picking
- you add a product with a 'non-labelable' UOM (e.g. kilos)
- you mark as todo
- you change the quantity to be more than the demand
- you print labels

Results: you have 2 labels instead of 1 because there are 2 stock move lines, 1 line for the initial demand and 1 line for the excess.
https://drive.google.com/file/d/1aIHgxKHSDaDZiSoYx_Ush9-LvaYZeGQD/view?usp=drive_link

Reply:
Hello @Sehnde 

As discussed with @ansh-odoo and @thomasdeleval,

len(line) will not always be 1 and if labellable is false then it will be 1 per move line.

Thanks.

Reply:
When you iterate over a list with a _for_ loop, it will always process **_1_** element at a time.
So `for line in self.move_ids.move_line_ids:` means: for **each** line in that list.
`line` is a single element in `self.move_ids.move_line_ids` so `len(line)` will **always** be 1